mypyc/primitives/struct_regsitry.py

```diff
@@ -0,0 +1,24 @@
+"""Struct registries for C backend"""
+
+from typing import List, NamedTuple
+from mypyc.ir.rtypes import RType
+
+CStructDescription = NamedTuple(
+    'CStructDescription', [('name', str),
+                           ('names', List[str]),
+                           ('types', List[RType])])
+
+
+def c_struct(name: str,
+             names: List[str],
+             types: List[RType]) -> CStructDescription:
+    """Define a known C struct for generating IR to manipulate it
+
+    name: The name of the C struct
+    types: type of each field
+    names: name of each field
+           TODO: the names list can be empty in the future when we merge Tuple as part of Struct
+    """
+    return CStructDescription(name, names, types)
```

Comment:
This doesn't actually add anything to a registry, it just constructs a description?

Reply:
yes, for now, I haven't figure out what's the best way to use the description, should we use it as a one-shot(like custom op) or add to a registry for matching. I think it's OK to postpone that to when we finally implement the `len` primitive